### PR TITLE
Initial lab-internal Nocturne version

### PR DESCRIPTION
**Pull request by daphnecor**
## Description

This pull request merges the `nocturne_fork_cleanup` branch into `develop`. The `nocturne_fork_cleanup` branch contains several changes that clean up the codebase and remove unused code. The resulting repository is the version that should be used internally in the EMERGE lab for further development of Nocturne.

## Changes Made

The following changes were made in the `nocturne_fork_cleanup` branch:

- [x] Clean-up:
    - [x] Removed `algos` directory
    - [x] Removed `examples` directory
    - [x] Removed configurations related to `algos` and `examples`
    - [x] Removed directory for cluster scripts
- [x] Dependencies and `poetry`:
    - [x] Added `poetry` for and to simplify package management
    - [x] Removed unused dependencies (after removal of `algos` and `examples`)
    - [x] Split and updated dependencies for usage and development
- [ ] Code quality and `pre-commit`:
   - [x] Added `pre-commit`
   - [x] Added code quality utilities for development (e.g.,`black`, `flake8`, `isort`, `pydocstyle`, and `pylint`)
   - [ ] Applied code quality utilities to all files in code base (draft PR: https://github.com/Emerge-Lab/nocturne_lab/pull/2)
- [ ] Documentation (PR ready for review: https://github.com/Emerge-Lab/nocturne_lab/pull/3):
   - [ ] Updated documentation for lab version of package
   - [ ] Added hosted documentation
- [ ] <s>Publish package</s> -> Moved forward
   - [ ] <s>Publish package to PyPI</s>
- [x] Examples and tutorials
   - [x] Minimal base environment
   - [x] Minimal PPO example
- [ ] _(Optional)_ <s>Support for multiple Python versions</s> -> Moved forward
- [ ] _(Optional)_ <s>Update legacy Github workflows</s> -> Moved forward
- [ ] _(Optional)_ <s>Fix notebook editable package bug</s> -> Moved forward

## Related Issues

This pull request resolves the no outstanding issues.

## Review checklist
_TBA_

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -1,9 +1,4 @@
-# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
```
Comment (eugenevinitsky) [inline build.py:1]: Unfortunately you can't remove an MIT license from

File: build.py [hunk 2]
```diff
@@ -12,18 +7,14 @@
...
-from setuptools import Extension, setup
-from setuptools.command.build_ext import build_ext
-
-# Reference:
-# https://www.benjack.io/2017/06/12/python-cpp-tests.html
 
+logging.basicConfig(level=logging.INFO)
 
-class CMakeExtension(Extension):
+class CMakeExtension(Pybind11Extension):
```
Comment (eugenevinitsky) [inline build.py:13]: Nice

File: examples/02_nocturne_concepts.ipynb [hunk 3]
```diff
@@ -0,0 +1,785 @@
...
+    "Calling `scenario.visible_state()` returns a dictionary with four matrices:\n",
+    "- `stop_signs`: The visible stop signs \n",
+    "- `traffic_lights`: The states for the traffic lights from the perspective of the ego driver(red, yellow, green).\n",
+    "- `road_points`: The observable road points (static elements in the scene).\n",
+    "- `objects`: The observable road objects (vehicles, pedestrians and cyclists).\n",
+    "\n",
+    "<figure>\n",
+    "<center>\n",
+    "<img src='https://drive.google.com/uc?id=1fG43NvPCzaimmW99asRdB73qY-F4u-q0' width='700'/>\n",
+    "<figcaption>To investigate coordination under partial observability, agents in Nocturne can only see an obstructed view of their environment. In this simplified traffic scene, we construct the state for the red ego driver. Note that Nocturne assumes that stop signs can be viewed, even if they are behind another driver. </figcaption></center>\n",
```
Comment (eugenevinitsky) [inline examples/02_nocturne_concepts.ipynb:314]: There's a figure here that doesn't load for me

**Review by eugenevinitsky — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -1,235 +1,146 @@
...
+obs_dict = env.reset()
+num_agents = len(env.controlled_agents)
 
-More videos can be found [here](https://www.nathanlct.com/research/nocturne).
+# Step through env
+for _ in range(1000):
 
-The corresponding paper is available at: [https://arxiv.org/abs/2206.09889](https://arxiv.org/abs/2206.09889). Please cite the paper and not the GitHub repository, using the following citation:
+    # Take action(s)
+    action_dict = {
```
[the remark this answers is not in the document]
Comment (daphnecor) [reply README.md:23]: Good catch. Edited the example.

File: build.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (daphnecor) [reply build.py:1]: Yes, we have a LICENSE file in the root dir with the correct information.

**Pull request merged**